src/collectors/vm/prometheus_vm_memory_collector.erl

```diff
@@ -27,8 +27,9 @@ register() ->
 register(Registry) ->
   ok = prometheus_registry:register_collector(Registry, ?MODULE).
 
-deregister(_) -> ok.
+deregister(_) -> true.
```

Comment:
This was the path of least resistance to appease dialyzer. The other modules that actually implement `deregister/1` end up with a call to `ets:match_delete/2` which has the type:

```erlang
-type match_pattern() :: atom() | tuple().
-type tab() :: atom() | tid().
%% -type tid() :: "A table identifier, as returned by new/2".
-spec match_delete(Tab :: tab(), Pattern :: match_pattern()) -> true.
```